discord/poll.py

```diff
@@ -0,0 +1,641 @@
+            question=question,
+        )
+        self._answers = [PollAnswer(data=answer, message=message, poll=self) for answer in data['answers']]
+        self._message = message
+        self._state = state
+        self._expiry = expiry
+
+        try:
+            self._results = data['results']
+        except KeyError:
+            self._results = None
+
+        return self
+
+    def _to_dict(self) -> PollCreatePayload:
+        data: PollCreatePayload = {
+            'allow_multiselect': self.multiselect,
+            'question': self._question_media.to_dict(),
+            'duration': self._hours_duration,
+            'layout_type': self.layout_type.value,
+            'answers': [{'poll_media': answer._to_dict()} for answer in self.answers],
+        }
+        return data
+
+    def __str__(self) -> str:
+        return self._question_media.text
+
+    def __repr__(self) -> str:
+        return f"<Poll duration={self.duration} question=\"{self.question}\" answers={self.answers}>"
+
+    def __len__(self) -> int:
+        return len(self.answers)
+
+    @property
+    def question(self) -> str:
+        """:class:`str`: Returns this poll answer question string."""
+        return self._question_media.text
+
+    @property
+    def emoji(self) -> Optional[PartialEmoji]:
+        """Optional[:class:`PartialEmoji`]: Returns the emoji for this poll's question."""
+
+        return None  # As of now, polls questions don't support emojis
+
+    @property
+    def answers(self) -> List[PollAnswer]:
+        """List[:class:`PollAnswer`]: Returns a read-only copy of the answers"""
+        return self._answers.copy()
+
+    @property
+    def expiry(self) -> Optional[datetime.datetime]:
+        """Optional[:class:`datetime.datetime`]: A datetime object representing the poll expiry.
+
+        .. note::
+
+            This will **always** return ``None`` if the poll is not part of a message.
+        """
+        return self._expiry
+
+    @property
+    def created_at(self) -> Optional[datetime.datetime]:
+        """:class:`datetime.datetime`: Returns the poll's creation time, or ``None`` if user-created."""
+
+        if not self._message:
+            return
+        return self._message.created_at
+
+    @property
+    def message(self) -> Optional[Message]:
+        """:class:`Message`: The message this poll is from."""
+        return self._message
+
+    @property
+    def total_votes(self) -> int:
+        """:class:`int`: Returns the sum of all the answer votes."""
+        return sum([answer.vote_count for answer in self.answers])
+
+    def is_finalized(self) -> bool:
+        """:class:`bool`: Returns whether the poll has finalized.
+
+        It always returns ``False`` if the poll is not part of a
+        fetched message. You should consider accessing this method
+        via :attr:`Message.poll`
+        """
+        if not self._results:
+            return False
+        return self._results['is_finalized']
```

Comment:
I'm not sure why you're not just storing the state as a bool, storing the result payload seems like a waste.

Reply:
Yeah, this is now updated in the latest commits.